<!DOCTYPE html>
<html lang="en-us">
  <head>


    <meta charset="UTF-8">
    <title>IFNL-Evolution by giffordlabcvr</title>
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <link rel="stylesheet" type="text/css" href="website/stylesheets/normalize.css" media="screen">
    <link href='https://fonts.googleapis.com/css?family=Open+Sans:400,700' rel='stylesheet' type='text/css'>
    <link rel="stylesheet" type="text/css" href="website/stylesheets/stylesheet.css" media="screen">
    <link rel="stylesheet" type="text/css" href="website/stylesheets/github-light.css" media="screen">


  </head>
  
  <body>
  
    <section class="page-header">
      <h1 class="project-name">IFNL-Evolution</h1>
      <h2 class="project-tagline">Comparative evolutionary analysis of IFNL genes</h2>
      <a href="https://github.com/giffordlabcvr/IFNL-Evolution/zipball/master" class="btn">Download</a>
      <a href="https://github.com/giffordlabcvr/IFNL-Evolution" class="btn">View on GitHub</a>
    </section>

    <section class="main-content">


	   <h3>
		 <a id="Background" class="anchor" href="#Background" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a><strong>Comparative analysis of lambda interferons using GLUE</strong>
	   </h3>
	   <hr>



	   <p>

		 <a target="_blank" href="https://pfam.xfam.org/family/IL28A"></a><b>Lambda interferons (IFNLs)</b></a> are inflammatory
		 <a target="_blank" href="https://en.wikipedia.org/wiki/Cytokine"><b>cytokines</b></a>
		 that modulate immune responses during acute viral infection and tissue inflammation.


	   </p>


         <p><img src="./website/images/ifnl-header.jpg" alt="Cytokine release" /></p>
         

<blockquote>
		  Following microbial infection, IFNLs are abundantly expressed at mucosal interfaces
		  such as the respiratory and gastrointestinal tracts 
</blockquote>
		 



	  <h3>
		<a id="ifnl-glue" class="anchor" href="#ifnl-glue" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a><strong>Description</strong>
	  </h3>
      <hr>

		<p>
		  This <a target="_blank" href="https://github.com/">GitHub</a> repository contains
		  data and program logic supporting comparative and phylogenetic investigations of
		  <b>IFNLs</b>.		
		</p>

		<p>
		  IFNL homologues have been identified in almost all vertebrate classes, however,
		  the understanding of IFNL biology is dominated by studies in mice and humans.	
		</p>


	  <br>     
	  <h3>
	  <a id="WhatGlueProject" class="anchor" href="#glueProject" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a><strong>What is a GLUE project?</strong>
	  </h3>
	  <hr>


		<p>
		   <a target="_blank" href="http://glue-tools.cvr.gla.ac.uk/"><b>GLUE</b></a> is an open, integrated
		   software toolkit that provides functionality for storage and interpretation of
		   sequence data. 
		 </p>

		 <p>
		   <b>GLUE</b> supports the development of <b>“projects”</b> containing the data items
		   required for comparative genomic analysis 
		   (e.g.  sequences, multiple sequence alignments, genome feature annotations,
		   and other sequence-associated data). 
		 </p>

		 <p>
		   Projects are loaded into the <b>GLUE "engine"</b>, creating a relational database
		   that represents the semantic relationships between data items.
		   This provides a robust foundation for the implementation of <b>systematic
		   comparative analyses</b> and the development of <b>sequence-based resources</b>. 
		 </p>

		 <p>		 
			The core schema of this database can be extended to accommodate
			the idiosyncrasies of different projects, and
			GLUE provides a scripting layer (based on JavaScript) for developing custom
			analysis tools. 
		 </p>

		 <p>
			Hosting of GLUE projects in an online version control system (e.g. GitHub) provides
			a mechanism for their stable, collaborative development.
		 </p>
	   

		 <p>
		   Some examples of '<b>sequence-based resources</b>' built for viruses using GLUE include:
		 </p>
		   
		   

		 <p>

		 <ul>
		 		 
		   <li> 
			 <a target="_blank" href="http://cov-glue.cvr.gla.ac.uk/"><b>COV-GLUE</b></a>: 
			 A GLUE resource for tracking genetic variation in SARS-COV2.
			 CoV-GLUE contains a database of amino acid replacements, insertions and
			 deletions which have been observed in GISAID hCoV-19 sequences sampled from the pandemic
		   </li>
		   <br>
		   
		   <li>
			 <a target="_blank" href="http://rabv-glue.cvr.gla.ac.uk/"><b>RABV-GLUE</b></a>:
			 Tailored toward epidemiological tracking of rabies virus (RABV).
			 Includes a database of RABV sequences and metadata from NCBI, updated daily and arranged into major and minor clades, and
			 an analysis tool providing genotyping, analysis and visualisation of submitted FASTA sequences.
 		   </li>
 		   <br>
		   
		   <li>
			 <a target="_blank" href="http://hcv-glue.cvr.gla.ac.uk/"><b>HCV-GLUE</b></a>:
			 This GLUE resource aims to support analysis of drug resistance and vaccine
			 escape in hepatitis C virus (HCV).
			 A database of HCV sequences and metadata from NCBI, updated daily and arranged
			 into clades (genotypes, subtypes). As well as pre-built multiple-sequence
			 alignments of NCBI sequences, it includes an analysis tool providing genotyping,
			 drug resistance analysis and visualisation of submitted FASTA sequences.
           </li>
	   
		 </ul>

		   
		 </p>
	   


       <br>
	   <h3>
	   <a id="WhyGlueProject" class="anchor" href="#glueProject" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a><strong>What does building the IFNL-GLUE project offer?</strong>
	   </h3>
       <hr>

		 
 
		 <p>
		 
			IFNL-GLUE offers a number of advantages for performing comparative sequence
			analysis of hepadnaviruses:

		   <ol>

			 <li> <strong>Reproducibility</strong>. 
			 For many reasons, bioinformatics analyses are notoriously difficult to reproduce.
			 The GLUE framework supports the implementation of fully reproducible 
			 comparative genomics through the introduction of data standards and the use 
			 of a relational database to capture the semantic links between data items.
			 </li>
			 
			 <li> <strong>Making data objects and analysis logic reusable</strong>. 
			 For many - if not most - comparative genomic analyses, data preparation is nine
			 tenths of the battle. The GLUE framework has been designed to ensure that 
			 work spent preparing high-value data items such as multiple sequence alignments
			 need only be performed once. Hosting of GLUE projects in an online version control 
			 system such as GitHub allows for collaborative management of important data items
			 and community testing of hypotheses.
			 </li>

			 <li> <strong>Validation</strong>. 
			 Building GLUE projects entails mapping the semantic links between data items
			 (e.g. sequences, tabular data, multiple sequence alignments).
			 This process provides an opportunity 
			 for cross-validation, and thereby enforces a high level of data integrity.
			 </li>
			 
			 <li> <strong>Standardisation of the genomic co-ordinate space</strong>. GLUE
			 projects allow all sequences to utilise the coordinate space of a chosen 
			 reference sequence. Contingencies associated with insertions and deletions
			 (indels) are handled in a systematic way.
			 </li>
			 
			 <li> <strong>Predefined, fully annotated reference sequences</strong>:
			 This project includes fully-annotated reference sequences for major lineages
			 within the <i>IFNL</i> family.
			 </li>
			 
			 <li> <strong>Alignment trees</strong>:
			 GLUE allows linking of alignments constructed at distinct taxonomic levels
			 via an <b>alignment tree</b>
			 data structure. In the alignment tree, each alignment
			 is constrained to a standard reference sequence, thus all multiple sequence
			 alignments are linked to one another via a standardised coordinate system.
			 </li>
			 
		   </ol>

		 </p>







       <br>
	   <h3>
	   <a id="glueProject" class="anchor" href="#glueProject" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a><strong>GLUE project</strong>
	   </h3>
       <hr>
      	   
	   <p>
	
         On computers with <a href="http://glue-tools.cvr.gla.ac.uk/#/installation">GLUE installed</a>,
         the <b>IFNL-Evolution</b> project can be instantiated by navigating to the project folder,
         initiating GLUE, and issuing the following command in the GLUE shell:
         
  <pre>
  <code>Mode path: /
  GLUE> run file ifnlProject.glue</code></pre>
		 
	   </p>



       <br>
	   <h3>
		 <a id="Publications" class="anchor" href="#Publications" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a><strong>Related Publications</strong>
	   </h3>
       <hr>

		  <p>
		

		  Bamford CGG, Aranday-Cortes E, Filipe IC, Sukumar S, Mair D, Filipe ADS, Mendoza JL,
		  Garcia KC, Fan S, Tishkoff SA, McLauchlan J.
		  <b>(2018)</b>
		  <br>
		  A polymorphic residue that attenuates the antiviral potential of interferon lambda 4 in hominid lineages.
		  <br>
		  <i>PLoS Pathogens</i>
		  <a href="https://journals.plos.org/plospathogens/article?id=10.1371/journal.ppat.1007307" target="_blank">[view]</a> 
		  <br>
		  <br>

		  Fox BA, Sheppard PO, and PJ O'Hara 
		  <b>(2018)</b>
		  <br>
		  The role of genomic data in the discovery, annotation and evolutionary interpretation of the interferon-lambda family.
		  <br>
		  <i>PLoS One</i>
		  <a href="https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0004933" target="_blank">[view]</a> 
		  <br>
		  <br>

		  Singer JB, Thomson EC, McLauchlan J, Hughes J, and RJ Gifford
		  <b>(2018)</b>
		  <br>
		  GLUE: A flexible software system for virus sequence data.
		  <br>
		  <i>BMC Bioinformatics</i>
		  <a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-018-2459-9" target="_blank">[view]</a> 
		  <br>
		  <br>

		  Zhu H, Dennis T, Hughes J, and RJ Gifford
		  <b>(2018)</b>
		  <br>
		  Database-integrated genome screening (DIGS): exploring genomes heuristically using sequence similarity search tools and a relational database.
		  <a href="https://www.biorxiv.org/content/early/2018/04/25/246835" target="_blank">[preprint]</a> 
		  <br>
		  <br>

		  Gifford RJ, Blomberg B, Coffin JM, Fan H, Heidmann T, Mayer J, Stoye J, Tristem M, and WE Johnson
		  <b>(2018)</b>
		  <br>
		  Nomenclature for endogenous retrovirus (ERV) loci.
		  <br>
		  <i>Retrovirology</i>
		  <a href="https://retrovirology.biomedcentral.com/articles/10.1186/s12977-018-0442-1" target="_blank">[view]</a> 
		  <br>
		  <br>
  	   
 
		  Krause CD.
		  <b>(2016)</b>
		  <br>
		  Intron loss in interferon genes follows a distinct set of stages, and may confer an evolutionary advantage.
		  <br>
		  <i>Cytokine</i>
		  <a href="https://pubmed.ncbi.nlm.nih.gov/27155818/" target="_blank">[view]</a> 
		  <br>
		  <br>


		  Katzourakis A. and RJ. Gifford
		  <b>(2010)</b>
		  <br>
		  Endogenous viral elements in animal genomes.
		  <br>
		  <i>PLoS Genetics</i>
		  <a href="http://www.plosgenetics.org/article/info:doi/10.1371/journal.pgen.1001191" target="_blank">[view]</a>

		  </p>







       <br>
	   <h3>
		 <a id="contributors" class="anchor" href="#contributors" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a><strong>Contributors</strong>
	   </h3>
       <hr>

	   


		 <p><a href="https://giffordlabcvr.github.io/virus-evolution/index.html" target="_blank">Robert J. Gifford</a> (<a href="mailto:robert.gifford@glasgow.ac.uk">robert.gifford@glasgow.ac.uk</a>)</p>
		 <p>Connor Bamford (<a href="mailto:connor.bamford@glasgow.ac.uk">c.bamford@qub.ac.uk</a>)</p>
		 

        <p>
        <br>


	   <h3>
		 <a id="license" class="anchor" href="#license" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a><strong>License</strong>
	   </h3>
       <hr>

		 <p>
		   This project is licensed under the <a href="https://www.gnu.org/licenses/agpl-3.0.en.html">GNU Affero General Public License v. 3.0</a>.
		 </p>

	   <br>



        <br>
        <br>
        <br>
        <br>
        <img src="./website/images/ifnl-footer-left.jpg" alt="Catshark" />

        </p>



	  <footer class="site-footer">
		<span class="site-footer-owner"><a href="https://github.com/giffordlabcvr/IFNL-Evolution">IFNL-Evolution</a> is maintained by <a href="https://github.com/giffordlabcvr">giffordlabcvr</a>.</span>

		<span class="site-footer-credits">This page was generated by <a href="https://pages.github.com">GitHub Pages</a> using the <a href="https://github.com/jasonlong/cayman-theme">Cayman theme</a> by <a href="https://twitter.com/jasonlong">Jason Long</a>.</span>
	  </footer>

    </section>


  </body>
</html>
